core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -672,6 +685,16 @@ object Parser {
     */
   sealed abstract class Expectation {
     def offset: Int
+
+    /** This is a reverse order stack (most recent context first)
+      * of this parsing error
+      */
+    def context: List[String] =
```

Comment:
Can we make this `tailrec`? It'd probably require a `reverse` at the end but maybe it's worth pursuing?

Reply:
Seems to me if your errors are so nested you blow the stack here you will have other issues. I don't think it is worth a cost to retain the ability of having 50k deep nested contexts.

What do you think?

Reply:
Fair. Let's leave it ;)